components/operator/internal/controllers/stack/storage/es/function.go

```diff
@@ -0,0 +1,71 @@
+package es
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+
+	"github.com/formancehq/operator/apis/stack/v1beta3"
+	"github.com/go-logr/logr"
+)
+
+const (
+	stacksIndex = "stacks"
+)
+
+func DropESIndex(config *v1beta3.ElasticSearchConfig, logger logr.Logger, stackName string) error {
+	client, err := NewElasticSearchClient(config)
+	if err != nil {
+		logger.Error(err, "ES client error")
+		return err
+	}
+
+	var (
+		buf bytes.Buffer
+		res Response
+	)
+
+	query := map[string]interface{}{
```

Comment:
I think you can replace all this stuff by some method provided by the ES library

Reply:
```suggestion
func DropESIndex(config *v1beta3.ElasticSearchConfig, logger logr.Logger, stackName string) error {
	client, err := NewElasticSearchClient(config)
	if err != nil {
		logger.Error(err, "ES client error")
		return err
	}

	response, err := client.Delete(stacksIndex, stackName)

	if err != nil {
		logger.Error(err, "ELK: Error during index drop")
		return err
	}

	defer response.Body.Close()
	if response.StatusCode > 300 {
		logger.Error(fmt.Errorf("ELK: Status code: %d", response.StatusCode), "ELK: Error during index drop")
		return fmt.Errorf("ELK: Status code: %d", response.StatusCode)
	}

	logger.Info(fmt.Sprintf("ELK Status code: %d", response.StatusCode))

	var res Response
	if err := json.NewDecoder(response.Body).Decode(&res); err != nil {
		logger.Error(err, "ELK: Error decoding es body")
		return err
	}

	if response.StatusCode == 200 {
		logger.Info("ES Index Dropped")
	}

	return nil
}
```suggestion

```
```

Reply:
```
func DropESIndex(config *v1beta3.ElasticSearchConfig, logger logr.Logger, stackName string) error {
	client, err := NewElasticSearchClient(config)
	if err != nil {
		logger.Error(err, "ES client error")
		return err
	}

	response, err := client.Delete(stacksIndex, stackName)

	if err != nil {
		return err
	}

	if response.IsError() {
		return fmt.Errorf("ELK status: %d", response.StatusCode)
	}

	return nil
}
```